### PR TITLE
[eclipse/xtext-core#1223] adapted documentation snippets to use Java  instead of Xtend to align with wizard

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext-core#1223] adapted documentation snippets to use Java  instead of Xtend to align with wizard

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
TODO check if the got everything right and did not overlook anything

File: xtext-website/documentation/303_runtime_concepts.md [hunk 1]
```diff
@@ -249,6 +252,26 @@ override getScope(EObject context, EReference reference) {
...
 ```
 
+or implemented in Java
+
+```java
+@Override
+public IScope getScope(EObject context, EReference reference) {
+    // We want to define the Scope for the Element's superElement cross-reference
+    if (context instanceof Element
+            && reference == MyDslPackage.Literals.ELEMENT__SUPER_ELEMENT) {
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: .equals?

File: xtext-website/documentation/303_runtime_concepts.md [hunk 2]
```diff
@@ -249,6 +252,26 @@ override getScope(EObject context, EReference reference) {
 }
 ```
 
+or implemented in Java
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: Why not just remove above Xtend code?

File: xtext-website/documentation/303_runtime_concepts.md [hunk 3]
```diff
@@ -269,6 +292,21 @@ override getScope(EObject context, EReference reference) {
 }
 ```
 
+```java
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: Why not remove above xtend code?

File: xtext-website/documentation/303_runtime_concepts.md [hunk 4]
```diff
@@ -269,6 +292,21 @@ override getScope(EObject context, EReference reference) {
...
+    @Override
+    public IScope getScope(final EObject context, final EReference reference) {
+        if (context instanceof Element
+                && reference == MyDslPackage.Literals.ELEMENT__SUPER_ELEMENT) {
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: .equals?

File: xtext-website/documentation/303_runtime_concepts.md [hunk 5]
```diff
@@ -269,6 +292,21 @@ override getScope(EObject context, EReference reference) {
...
+            EObject rootElement = EcoreUtil2.getRootContainer(context);
+            List<Element> candidates = EcoreUtil2.getAllContentsOfType(rootElement, Element.class);
+            IScope existingScope = Scopes.scopeFor(candidates);
+            // Scope that filters out the context element from the candidates list
+            return new FilteringScope(existingScope, (e) -> e.getEObjectOrProxy() != context);
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: !.equals()?

File: xtext-website/documentation/303_runtime_concepts.md [hunk 6]
```diff
@@ -697,11 +735,12 @@ In the Eclipse UI scenario, when there is a workspace, users will expect the enc
...
-override configureRuntimeEncodingProvider(Binder binder) {
-    binder.bind(IEncodingProvider)
-          .annotatedWith(DispatchingProvider.Runtime)
-          .to(MyEncodingProvider)
+```java
+@Override
+public void configureRuntimeEncodingProvider(Binder binder) {
+    binder.bind(IEncodingProvider.class)
+    .annotatedWith(DispatchingProvider.Runtime.class)
+    .to(MyEncodingProvider.class);
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: These used to be indented

File: xtext-website/documentation/303_runtime_concepts.md [hunk 7]
```diff
@@ -815,18 +854,19 @@ If in addition to the main language your tests require using other languages for
...
+    @Override
+    protected Injector internalCreateInjector() {
+        MyOtherLangLanguageStandaloneSetup.doSetup();
+        return super.internalCreateInjector();
     }
 }
 
 // @RunWith(XtextRunner) // JUnit 4
 @ExtendWith(InjectionExtension) // JUnit 5
 @InjectWith(MyLanguageWithDependenciesInjectorProvider)
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: `.class`?

File: xtext-website/documentation/303_runtime_concepts.md (hunk 1 above)
Comment (szarnekow) [reply xtext-website/documentation/303_runtime_concepts.md]: The EPackage contents object are singletons so identity equals is what I would consider idiomatic usage of the Literals interface.

File: xtext-website/documentation/303_runtime_concepts.md (hunk 2 above)
Comment (szarnekow) [reply xtext-website/documentation/303_runtime_concepts.md]: 👍

File: xtext-website/documentation/310_eclipse_support.md [hunk 8]
```diff
@@ -117,21 +117,22 @@ It is important to know that, for a given offset in a model file, many possible
...
+public void completeEvent_Code(
   Event event, Assignment assignment, 
   ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
   // call implementation of superclass
-  super.completeEvent_Code(model, assignment, context, acceptor)
+  super.completeEvent_Code(model, assignment, context, acceptor);
 
   // compute the plain proposal
-  val String proposal = event.getName() + "ID"
+  final String proposal = event.getName() + "ID"
```
Comment (szarnekow) [inline xtext-website/documentation/310_eclipse_support.md]: Unnecessary noise. `final` can be dropped here.

**Comment by cdietrich** [comment]
@tivervac please re-review

File: xtext-website/documentation/303_runtime_concepts.md (hunk 1 above)
Comment (tivervac) [reply xtext-website/documentation/303_runtime_concepts.md]: Alright, make sense

File: xtext-website/documentation/303_runtime_concepts.md (hunk 4 above)
Comment (tivervac) [reply xtext-website/documentation/303_runtime_concepts.md]: As Sebastian said above, not really needed

File: xtext-website/documentation/303_runtime_concepts.md [hunk 9]
```diff
@@ -255,18 +260,20 @@ The [MapBasedScope]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xt
...
-            && reference == MyDslPackage.Literals.ELEMENT__SUPER_ELEMENT) {
-        val rootElement = EcoreUtil2.getRootContainer(context)
-        val candidates = EcoreUtil2.getAllContentsOfType(rootElement, Element)
-        val existingScope = Scopes.scopeFor(candidates)
-        // Scope that filters out the context element from the candidates list
-        return new FilteringScope(existingScope, [getEObjectOrProxy != context])
+
+```java
+    @Override
+    public IScope getScope(final EObject context, final EReference reference) {
```
Comment (tivervac) [inline xtext-website/documentation/303_runtime_concepts.md]: No real need for the finals here

**Pull request merged**